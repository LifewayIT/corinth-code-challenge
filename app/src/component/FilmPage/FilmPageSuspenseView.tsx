import React, {FunctionComponent} from 'react';
import splashScreen from '../../assets/image/long-time-ago.png';

export const FilmPageSuspenseView: FunctionComponent = () => (
    <div>
        <img src={splashScreen}/>
    </div>
);
