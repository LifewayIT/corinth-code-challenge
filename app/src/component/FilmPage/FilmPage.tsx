import {Grid} from '@mui/material';
import React, {FunctionComponent} from 'react';
import {RouteComponentProps} from 'react-router';
import {useRecoilCallback, useRecoilValue} from 'recoil';
import {FilmAtom} from '../../graph/film/filmAtom';
import {FilmCard} from '../FilmCard/FilmCard';
import {ItemList} from '../ItemList/ItemList';
import {withErrorCatch} from '../withErrorCatch';
import {withSuspense} from '../withSuspense';
import {FilmPageSuspenseView} from './FilmPageSuspenseView';
import {BasePage} from "../BasePage/BasePage";

type FilmPageProps = RouteComponentProps<{ id: string }>;

const FilmPage: FunctionComponent<FilmPageProps> = ({match: {params}}) => {

    const film = useRecoilValue(FilmAtom.film);
    useRecoilCallback(({set}) =>
        (val: string) => set(FilmAtom.filmId, val)
    )(params.id);

    return (
        <BasePage show={Boolean(film)} card={film && (
            <FilmCard film={film}/>
        )}>
            {film && (
                <Grid container spacing={2} direction="column">
                    <Grid item xs>
                        <ItemList items={film.characters} title="Characters"/>
                    </Grid>
                    <Grid item xs>
                        <ItemList items={film.species} title="Species"/>
                    </Grid>
                    <Grid item xs>
                        <ItemList items={film.starships} title="Starships"/>
                    </Grid>
                    <Grid item xs>
                        <ItemList items={film.vehicles} title="Vehicles"/>
                    </Grid>
                </Grid>
            )}
        </BasePage>
    );

};

export default withErrorCatch(withSuspense(FilmPage, FilmPageSuspenseView));
