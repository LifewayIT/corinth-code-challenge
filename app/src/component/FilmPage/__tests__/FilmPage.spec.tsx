import React from 'react';
import {MemoryRouter, Route} from 'react-router';
import {create} from 'react-test-renderer';
import {RecoilRoot} from 'recoil';
import FilmPage from '../FilmPage';

jest.mock('../../../graph/film/filmService', () => ({
    getFilmById: () => Promise.resolve({
        id: '3',
        name: 'Return of the Jedi',
        episodeId: 6,
        openingCrawl: 'Luke Skywalker has returned to\r\nhis home planet of Tatooine in',
        director: 'Richard Marquand',
        producer: 'Howard G. Kazanjian, George Lucas, Rick McCallum',
        releaseDate: '1983-05-25',
        created: '2014-12-18T10:39:33.255000Z',
        edited: '2014-12-20T09:48:37.462000Z'
    })
}));

describe('FilmPage', () => {

    it('should change atom on page load', () => {

        const tree = create(
            <MemoryRouter>
                <RecoilRoot>
                    <Route path="*"
                           render={({match, history, location}) => (
                               <FilmPage match={{...match, params: {id: '3'}}} history={history} location={location}/>
                           )}/>
                </RecoilRoot>
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

});
