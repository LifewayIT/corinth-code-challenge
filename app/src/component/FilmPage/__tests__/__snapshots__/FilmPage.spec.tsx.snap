// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPage should change atom on page load 1`] = `
<div>
  <img
    src="test-file-stub"
  />
</div>
`;
